### PR TITLE
[MTDSA-719] Enable self-employment update.

### PR DESCRIPTION
**Pull request by GlassAndOneHalf**
- Implement new SelfEmploymentUpdate type that contains a subset of the fields contained in SelfEmployment.
- Add/modify tests.

### CODE REVIEW

File: public/api/conf/1.0/schemas/SelfEmployment_Update.json [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      "title": "Business description",
+      "description": "A 5 digit code that conforms to the SIC 2007 standard trade classifications",
+      "type": "string",
+      "pattern": "^[0-9]{5}$"
+    },
+    "businessAddressLineOne": {
+      "title": "Business address line one",
+      "description": "First line of the business address",
+      "type": "string"
+    },
```
Comment (dabd) [inline public/api/conf/1.0/schemas/SelfEmployment_Update.json]: Add Json validation keywords for businessAddressLine*? `{"minLength": 1,  "maxLength": 35`}

File: public/api/conf/1.0/schemas/SelfEmployment_Update.json [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    "businessAddressLineFour": {
+      "title": "Business address line four",
+      "description": "Fourth line of the business address",
+      "type": "string"
+    },
+    "businessPostcode": {
+      "title": "Business postcode",
+      "description": "Business postcode",
+      "type": "string"
+    }
```
Comment (dabd) [inline public/api/conf/1.0/schemas/SelfEmployment_Update.json]: Add Json validation keywords for businessPostcode? {"minLength": 1, "maxLength": 10}

**Comment by GlassAndOneHalf** [comment]
Updated JSON schema as per @dabd comments.

**Comment by dabd** [comment]
👍

**Pull request merged**